server/http/controller.go

```diff
@@ -0,0 +1,235 @@
+package http
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/kubeshop/tracetest/analytics"
+	"github.com/kubeshop/tracetest/executor"
+	"github.com/kubeshop/tracetest/openapi"
+	"github.com/kubeshop/tracetest/testdb"
+	"github.com/kubeshop/tracetest/tracedb"
+)
+
+type controller struct {
+	traceDB tracedb.TraceDB
+	testDB  testdb.Database
+	runner  executor.Runner
+}
+
+func NewController(traceDB tracedb.TraceDB, testDB testdb.Database, runner executor.Runner) openapi.ApiApiServicer {
+	return &controller{
+		traceDB: traceDB,
+		testDB:  testDB,
+		runner:  runner,
+	}
+}
```

Comment:
I'm ok with this now, but I think we should split this controller in multiple controllers in the future.

Reply:
totally agree, this is just a first step. This controller should be only a wrapper.